src/Tizen.NUI/src/public/Window/WindowEvent.cs

```diff
@@ -1082,5 +1083,108 @@ public void VisibiltyChangedSignalEmit(bool visibility)
+        }
+
+        private void OnAuxiliaryMessage(string key, string value, IntPtr optionsArray)
+        {
+            // at this case, cMemoryOwn is false, because this is just a test implementation, 
+            // so I create local Dali::Property::Array in dali adaptor's window.
+            // therefore, the PropertyArray needs be managed by Dali, and in nui, no need to free it.
+            using var tmp = new PropertyArray(optionsArray, false);
+            var size = tmp.Size();
+
+            List<string> tmpList = new List<string>();
+
+            for (int i = 0; i < size; i++)
+            {
+                string option = "none";
+                tmp.GetElementAt((uint)i).Get(out option);
+                tmpList.Add(option);
+            }
+
+            AuxiliaryMessageEventArgs e = new AuxiliaryMessageEventArgs();
+            e.Key = key;
+            e.Value = value;
+            e.Options = tmpList;
+
+            AuxiliaryMessageEventHandler?.Invoke(this, e);
+        }
+
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private delegate void AuxiliaryMessageEventCallbackType(string key, string value, IntPtr optionsArray);
+        private AuxiliaryMessageEventCallbackType AuxiliaryMessageEventCallback;
+        private event EventHandler<AuxiliaryMessageEventArgs> AuxiliaryMessageEventHandler;
+
+        /// <summary>
+        /// Auxiliary message is sent by displayer server when window's auxiliary was changed then display server sent the message.
+        /// When client application added the window's auxiliary hint and if the auxiliary is changed,
+        /// display server send the auxiliary message.
+        /// Auxiliary message has the key, value and options.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public event EventHandler<AuxiliaryMessageEventArgs> AuxiliaryMessage
+        {
+            add
+            {
+                if (AuxiliaryMessageEventHandler == null)
+                {
+                    AuxiliaryMessageEventCallback = OnAuxiliaryMessage;
+                    using var signal = new WindowAuxiliaryMessageSignal(this);
+                    signal.Connect(AuxiliaryMessageEventCallback);
+                }
+                AuxiliaryMessageEventHandler += value;
+            }
+            remove
+            {
+                AuxiliaryMessageEventHandler -= value;
+                if (AuxiliaryMessageEventHandler == null)
+                {
+                    if (AuxiliaryMessageEventCallback != null)
+                    {
+                        using var signal = new WindowAuxiliaryMessageSignal(this);
+                        signal.Disconnect(AuxiliaryMessageEventCallback);
+
+                        if (signal.Empty())
+                        {
+                            AuxiliaryMessageEventCallback = null;
+                        }
+                    }
+                }
+            }
+        }
+
+
+
```

Comment:
could you add the following in `DisconnectNativeSignals()`

            if (AuxiliaryMessageEventCallback != null)
            {
                using var signal = new WindowAuxiliaryMessageSignal(this);
                signal.Disconnect(AuxiliaryMessageEventCallback);
                AuxiliaryMessageEventHandler= null;
                AuxiliaryMessageEventCallback = null;
            }

thank you.